cpp/include/cuspatial/trajectory.hpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cudf/types.hpp>
+#include <memory>
+#include <rmm/mr/device/default_memory_resource.hpp>
+
+namespace cuspatial {
+namespace experimental {
+
+/**
+ * @brief Derive trajectories from points, timestamps, and object ids.
+ *
+ * Groups the input object ids to determine unique trajectories. Returns a
+ * table with the trajectory ids, the number of objects in each trajectory,
+ * and the offset position of the first object for each trajectory in the
+ * input object ids column.
+ *
+ * @param[in] x coordinates (km) (sorted by id, timestamp)
+ * @param[in] y coordinates (km) (sorted by id, timestamp)
+ * @param[in] object_id column of object (e.g., vehicle) ids
+ * @param[in] timestamp column (sorted by id, timestamp)
+ * @param[in] mr The optional resource to use for all allocations
+ *
```

Comment:
Missing @throw for the exceptions this can throw. Same for the other functions in this header.